sandstone/filewatcher.py

```diff
@@ -0,0 +1,43 @@
+from watchdog.observers import Observer
+import watchdog
+
+class Filewatcher(watchdog.events.FileSystemEventHandler):
+    """
+    Starts a watchdog instance to watch over the filesystem
+    Handles events - CREATE, REMOVE,MOVE
+    """
+    watched_directories = []
+
+    def __init__(self):
+        print 'Watchdog instance started'
+
+    def add_directory_to_watch(self, directory):
+        if directory not in self.watched_directories:
+            self.watched_directories.append(directory)
+        # Add the watcher
+        observer = Observer()
+        observer.schedule(self, directory, recursive=False)
+        observer.start()
+
+    def remove_directory_to_watch(self, directory):
+        if directory in self.watched_directories:
+            self.watched_directories.remove(directory)
+        # TODO remove the watchers if needed
```

Comment:
Looking through the documentation, we could probably work with a single observer, which has multiple watches. 
So, this is what we could do:
- Have a `_observer` class variable, which we instantiate on the `__init__` method of the class
- Have a `dict` of watches (returned by the `schedule` method on the observer), referenced by the directory. While adding a directory to watch, we add an entry in the dict. On removing a directory, we can remove the entry from the dict, and call the `unschedule` method on the observer, passing in the watch as a parameter.

How does this sound?